internal/ignition/ignition.go

```diff
@@ -445,6 +458,38 @@ func (g *installerGenerator) Generate(ctx context.Context, installConfig []byte,
 	return nil
 }
 
+func (g *installerGenerator) applyInfrastructureCRPatch(ctx context.Context) error {
+	log := logutil.FromContext(ctx, g.log)
+
+	// We are only patching the InfrastructureCR if the hosts count is 4
+	// and the three masters are schedulable.
+	if len(g.cluster.Hosts) != 4 {
```

Comment:
it will include disconnected no? We should be sure it counts only those that will be installed

Reply:
Should not we check that masters are schedulable too?

Reply:
> it will include disconnected no? We should be sure it counts only those that will be installed

Oh, I didn't know this. Can you tell me what's the right way to get the number of connected hosts? 

> Should not we check that masters are schedulable too?

openshift-installer takes care of setting the masters as schedulable. We could do it ourselves but this is already the default behavior in the installer for the 3+1 case.